catalogue/config.py

```diff
@@ -0,0 +1,122 @@
+
+    Applies the following checks, all of which are needed for the file to be
+    validated:
+
+    - config file can be read as a dictionary
+    - argument keys are a subset of (`--input_data`, `--code`, `--output_data`, `-csv`
+    `catalogue_results`)
+    - csv options specify a csv file
+    - argument values are all strings
+
+    config_loc specifies the location of the config file. Commands that involve
+    existing config files (such as the parser), will only use a validated config file.
+
+
+    Parameters:
+    ------------
+    config_loc : str
+
+    Returns:
+    ---------
+    Boolean indicating if a given config file is validated
+
+    """
+
+    config_dict = read_config_file(config_loc)
+
+    valid = True
+
+    # check config_dict is in fact a dictionary. The other checks require it to be a dictionary
+    if isinstance(config_dict, dict):
+
+        # checks that the config keys are a subset of the correct ones
+        valid_keys = ['catalogue_results','code','csv','input_data','output_data']
+        if not set(config_dict.keys()).issubset(valid_keys):
+            valid = False
+            print('Config error: invalid keys present in the yaml file')
+
+        # check that any csv configurations have the .csv extension
+        if 'csv' in config_dict.keys() and config_dict['csv'] is not None:
+            filename = config_dict['csv']
+
+            #change to use os path
+            extension = filename[-4:]
+            if extension != '.csv':
+                valid = False
+                print('Config error: csv argument has an invalid extension')
+
+        # check that all config file keys only have string values (i.e. no nested)
+        values_list = list(config_dict.values())
+        for value in values_list:
+            if not isinstance(value, str) and value is not None:
+                valid = False
+                print('Config error: config files are not all strings')
+
+        # check that there are no double keys
+
+    else:
+        valid = False
+        print('Config error: yaml file cannot be read as a dictionary')
+
+    return valid
+
+
+
+def config(args):
+
+    """
+    The `catalogue engage` command.
+
+    The config command is used to generate config files that allow the user
+    to specify argument inputs in advance. These config arguments will be used
+    by the parser to parse the relevant arguments for the other commands.
+
+    The config command:
+        - Checks if there already exists a validated config file
+        - Creates a new config file using specified input options
+        - Saves the config file in the base repository under `catalogue_config.yaml`
+
+    Parameters:
+    ------------
+    args : obj
+        Command line input arguments (argparse.Namespace).
+
+    Returns:
+    ---------
+    None
+    """
+
+    if not os.path.isfile(CONFIG_LOC):
+        print('No previous config file found')
+        print('Creating config file')
```

Comment:
I think these are redundant since you say later you create a new config file no matter what the outcome is of this check. I think it is fine to just say it once when you create the new one.

Reply:
I've removed the second print. I think we should keep the first print since it allows the user to account for the possibility that a config file was found but was not valid.